web/webpack.config.js

```diff
@@ -59,17 +70,42 @@ const config = {
       },
       {
         test: /\.(woff2?|ttf|otf|eot|svg)$/,
-        loader: 'file-loader'
+        use: [
+          {
+            loader: 'file-loader',
+            options: {
+              name: '[name].[ext]',
+              outputPath: 'static/fonts'
+            }
+          }
+        ]
       },
       {
         test: /\.yaml$/,
         use: ['json-loader', 'yaml-loader']
       }
     ]
   },
+  // replaces "process.env.____" with the values defined in the actual
+  // environment at build time
   plugins: [
     new webpack.EnvironmentPlugin({
       API_URL: null
+    }),
+
+    // uses module hashs as IDs instead of numeric indices, so adding a new
+    // file to the app doesn't cause vendored output hash to change
+    new webpack.HashedModuleIdsPlugin(),
+
+    // Inject our app scripts into our HTML kickstarter
+    new HtmlWebpackPlugin({
+      minify: { removeComments: true },
+      template: 'src/index.html'
+    }),
+    new HtmlWebpackIncludeAssetsPlugin({
+      assets: ['legacy.css', 'app.css'],
+      append: true,
+      hash: true
```

Comment:
I tried to put the content hash in the CSS filenames, but then this plugin wouldn't find them.  I got this to work though: the filenames are constant, but they get a `?[hash]` appended when they're injected into the `index.html`.  Unfortunately the hash is for the build, so it'll change pretty much every time meaning we'll get no cacheing.  Maybe we shouldn't have this hash?  I'm not clear on the best way to do this one.

Reply:
Maybe we leave the hash in for now, while stuff is changing rapidly, and we take it out later when the styles are more stable?

Reply:
That's smart.  @hbillings is smart.